executor/new_distsql.go

```diff
@@ -133,9 +134,18 @@ func (e *TableReaderExecutor) Next() (Row, error) {
 
 // Open implements the Executor Open interface.
 func (e *TableReaderExecutor) Open() error {
-	kvRanges := tableRangesToKVRanges(e.tableID, e.ranges)
-	var err error
-	e.result, err = distsql.SelectDAG(e.ctx.GetClient(), goctx.Background(), e.dagPB, kvRanges, e.ctx.GetSessionVars().DistSQLScanConcurrency, e.keepOrder, e.desc, getIsolationLevel(e.ctx.GetSessionVars()), e.priority)
+	var builder requestBuilder
+	kvReq, err := builder.SetTableRanges(e.tableID, e.ranges).
+		SetDAGRequest(e.dagPB).
```

Comment:
Could we init `kvReq` with `e` directly in one single function?

Reply:
They are nearly the same but not identical...such as `builder.SetTableRanges()` and `builder.SetTableHandles()`, so they can't be written into one single function.
Through it's not one single function, you can view it as a single line:

```
builder.SetXXX().SetYYY().SetZZZ().Build()
```